relay-server/src/actors/project_cache.rs

```diff
@@ -696,23 +764,19 @@ impl ProjectCacheBroker {
         {
             let reservoir_counters = project.reservoir_counters();
 
-            let sampling_state = utils::get_sampling_key(managed_envelope.envelope())
+            let sampling_project_state = utils::get_sampling_key(managed_envelope.envelope())
                 .and_then(|key| self.projects.get(&key))
-                .and_then(|p| p.valid_state());
+                .and_then(|p| p.valid_state())
+                .filter(|state| state.organization_id == own_project_state.organization_id);
```

Comment:
Why was this changed?

Reply:
I think the code is way more clear this way, not related to PR but just following the principle of cleaning up things in the same module I'm working in as i go along.